pallets/liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,1323 @@
+	) -> Self {
+		Self {
+			accumulated_rewards: Default::default(),
+			accumulated_rpz: Default::default(),
+			paid_accumulated_rewards: Default::default(),
+			total_shares_z: Default::default(),
+			liq_pools_count: Default::default(),
+			id,
+			updated_at,
+			reward_currency,
+			yield_per_period,
+			planned_yielding_periods,
+			blocks_per_period,
+			owner,
+			incentivized_token,
+			max_reward_per_period,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebugNoBound, TypeInfo)]
+pub struct LiquidityPoolYieldFarm<T: Config> {
+	id: PoolId,
+	updated_at: PeriodOf<T>,
+	total_shares: Balance,
+	total_valued_shares: Balance,
+	accumulated_rps: Balance,
+	accumulated_rpz: Balance,
+	loyalty_curve: Option<LoyaltyCurve>,
+	stake_in_global_pool: Balance, //NOTE: may be replace with: total_valued_shares * multiplier
+	multiplier: PoolMultiplier,
+	nft_class: NftClassIdOf<T>,
+	canceled: bool,
+}
+
+impl<T: Config> LiquidityPoolYieldFarm<T> {
+	fn new(
+		id: PoolId,
+		updated_at: PeriodOf<T>,
+		loyalty_curve: Option<LoyaltyCurve>,
+		multiplier: u32,
+		nft_class: NftClassIdOf<T>,
+	) -> Self {
+		Self {
+			accumulated_rps: Default::default(),
+			accumulated_rpz: Default::default(),
+			stake_in_global_pool: Default::default(),
+			total_shares: Default::default(),
+			total_valued_shares: Default::default(),
+			canceled: false,
+			id,
+			updated_at,
+			loyalty_curve,
+			multiplier,
+			nft_class,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct LoyaltyCurve {
+	initial_reward_percentage: FixedU128,
+	scale_coef: u32,
+}
+
+impl Default for LoyaltyCurve {
+	fn default() -> Self {
+		Self {
+			initial_reward_percentage: FixedU128::from_inner(500_000_000_000_000_000), // 0.5
+			scale_coef: 100,
+		}
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct Deposit<T: Config> {
+	shares: Balance,
+	valued_shares: Balance,
+	accumulated_rps: Balance,
+	accumulated_claimed_rewards: Balance,
+	entered_at: PeriodOf<T>,
+}
+
+impl<T: Config> Deposit<T> {
+	fn new(shares: Balance, valued_shares: Balance, accumulated_rps: Balance, entered_at: PeriodOf<T>) -> Self {
+		Self {
+			entered_at,
+			shares,
+			valued_shares,
+			accumulated_rps,
+			accumulated_claimed_rewards: Default::default(),
+		}
+	}
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use crate::weights::WeightInfo;
+	use frame_system::pallet_prelude::{BlockNumberFor, OriginFor};
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config + TypeInfo + pallet_nft::Config<ClassType = primitives::nft::ClassType>
+	{
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Asset type
+		type CurrencyId: Parameter + Member + Copy + MaybeSerializeDeserialize + Ord + From<u32>;
```

Comment:
From<u32> is only needed so you can run benchmarking, correct ?

Reply:
it's not needed for now, removed